### PR TITLE
Digital Ocean Integration tests

### PR DESCRIPTION
**Pull request by pmarques**
##### SUMMARY

Integration tests for Digital Ocean.

**This is my first test module created so please, all the feedback is welcome**

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

- test/integration/targets/digital_ocean

##### ADDITIONAL INFORMATION

Modules under integration testing:

- digital_ocean_sshkey

### CODE REVIEW

**Comment by mattclay** [comment]
@pmarques Thanks for taking the time to create an integration test.

Since we don't have support for automatic provisioning of DigitalOcean credentials, there's no need for a cloud plugin. We won't be able to run the integration tests as part of CI.

Instead, put the test in the legacy test directory. You'll need to rebase your branch first, as the legacy tests were just relocated to the `test/legacy/` directory.

You'll probably need the following:

- discard everything except`test/integration/targets/digital_ocean/tasks/sshkey.yml`
- rename to `test/legacy/roles/digital_ocean_sshkey/tasks/main.yml`
- create a `test/legacy/digital_ocean.yml` playbook to run the test

**Comment by Akasurde** [comment]
@pmarques Are you still working on this ? 

needs_info

**Comment by pmarques** [comment]
Hi @Akasurde tbh I kinda stoped using DO, I still follow the module and try to help. After the last message from @mattclay I never had time to update this PR.
Do you have any plans to continue the tests, any suggestion or expectations? I can do what @mattclay said proposed to push the PR forward.

**Comment by pmarques** [comment]
Update, rebase was required to get **test/legacy** folder

**Comment by Akasurde** [comment]
cc @mattclay

**Comment by Akasurde** [comment]
@mattclay Can you comment on the new changes ?

**Review by mattclay — APPROVED** [review]
Changes look OK.

File: test/legacy/roles/test_digital_ocean/tasks/sshkey.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+- name: Verify that SSH key was created
+  assert:
+    that:
+      - "result.changed"
+
+- name: "Delete ssh key"
+  digital_ocean_sshkey:
+    state: "absent"
+    fingerprint: "{{ result.data.ssh_key.fingerprint }}"
+    oauth_token: "{{ digitalocean_oauth_token }}"
```
Comment (BondAnthony) [inline test/legacy/roles/test_digital_ocean/tasks/sshkey.yml:19]: Doesn't this need a `register: result` for the `Verify that SSH key was deleted` to assert correctly?

Comment (pmarques) [reply test/legacy/roles/test_digital_ocean/tasks/sshkey.yml:19]: True, also missing in **floating_ip.yml**. I pushed a fixed but I was wondering if I should re-use *result* or use another variable...

**Comment by pmarques** [comment]
Also added a Makefile entry to run this tests

**Comment by BondAnthony** [comment]
shipit

**Comment by Akasurde** [comment]
@mattclay Could you please provide your feedback on this ? Thanks in advance.

File: test/legacy/roles/test_digital_ocean/tasks/sshkey.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    name: test-key1
+    ssh_pub_key: "{{ dummy_ssh_pub_key }}"
+    oauth_token: "{{ digitalocean_oauth_token }}"
+  register: result
+
+- name: Verify that SSH key was created
+  assert:
+    that:
+      - "result.changed"
+
```
Comment (gundalow) [inline test/legacy/roles/test_digital_ocean/tasks/sshkey.yml:14]: Could you add the key again and ensure no change is reported

File: test/legacy/roles/test_digital_ocean/tasks/sshkey.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  digital_ocean_sshkey:
+    state: "absent"
+    fingerprint: "{{ result.data.ssh_key.fingerprint }}"
+    oauth_token: "{{ digitalocean_oauth_token }}"
+  register: result
+
+- name: Verify that SSH key was deleted
+  assert:
+    that:
+      - "result.changed"
```
Comment (gundalow) [inline test/legacy/roles/test_digital_ocean/tasks/sshkey.yml:25]: Could you delete the key again and ensure no change is reported

**Review by gundalow — COMMENTED** [review]
These can be done in a follow-up PR (by someone else) I believe @pmarques isn't using DO as much anymore and I wouldn't want to make them do even more.

**Comment by gundalow** [comment]
@pmarques Thank you for doing this especially since you aren't using DO much anymore.

Thanks, everybody else for the reviews

Merged into `devel`

**Pull request merged**